### ISSUE TITLE
PodamFactoryImpl: Support for XML Datatype instantiation.

### ISSUE THREAD

**Issue by patrodyne** [mtedone/podam#69]
Thank you for Podam. I use it in production to generate samples requests for web service APIs. These POJOs often contain `XmlGregorianCalendar` properties. Here is a patch for `PodamFactoryImpl` to return values when instantiation requires the use of the XML Datatype factory.

1) Here are the imports ...
```
import java.util.GregorianCalendar;
import javax.xml.datatype.DatatypeConfigurationException;
import javax.xml.datatype.DatatypeFactory;
import javax.xml.datatype.Duration;
import javax.xml.datatype.XMLGregorianCalendar;
```

2) A helper method to get a static instance of the XML DatatypeFactory...

```
// A factory to create objects that map XML to/from Java.
private static DatatypeFactory xmlDatatypeFactory;
public static DatatypeFactory getXmlDatatypeFactory()
{
    if ( xmlDatatypeFactory == null )
    {
        try
        {
            xmlDatatypeFactory = DatatypeFactory.newInstance();
        }
        catch (DatatypeConfigurationException e)
        {
            throw new RuntimeException("Podam factory cannot create XML factory.", e);
        }
    }
    return xmlDatatypeFactory;
}
```

3) In the `createNewInstanceForClassWithoutConstructors(...)` method, within the test for 'no constructors' and after the loop to find a 'getInstance()' method, add this code...

```
// Did we find a getInstance() method to create the return value? 
if ( retValue == null )
{
    // No return value, so far.
    // Can we use an XML DatatypeFactory to create an instance? 
    if ( XMLGregorianCalendar.class.isAssignableFrom(clazz) )
    {
        retValue = getXmlDatatypeFactory().newXMLGregorianCalendar(new GregorianCalendar());
    }
    else if ( Duration.class.isAssignableFrom(clazz) )
    {
        List<Annotation> annotations = new ArrayList<Annotation>();
        getLongValueWithinRange(annotations, null);
        retValue = getXmlDatatypeFactory().newDuration(0L);
    }
}
```

**And a suggestion:** Change the `log.info(...)` in `PodamFactoryImpl` to `log.debug(...)`. This would allow my SysAdmin to set the log level to INFO instead of WARN without a flood of details from this class, IMHO.

**Comment by daivanov** [comment]
Fixed in PODAM 4.5.0.

**Comment by mtedone** [comment]
Released to Maven central

**Comment by daivanov** [comment]
Closing.

**Comment by daivanov** [comment]
Apparently `XMLGregorianCalendar` manufacturing test doesn't work properly.

**Comment by daivanov** [comment]
Fixed in PODAM 5.2.0.
